src/heliclockter/__init__.py

```diff
@@ -104,6 +115,33 @@ def _validate(cls: Type[DateTimeTzT], v: Any) -> Optional[DateTimeTzT]:
             dt = v if isinstance(v, _datetime.datetime) else parse_datetime(v)
             return cls.from_datetime(dt)
 
+    elif PYDANTIC_V2_AVAILABLE:
+
+        @classmethod
+        def __get_pydantic_core_schema__(cls, _: Any, __: GetCoreSchemaHandler) -> CoreSchema:
+            from_datetime_schema = core_schema.chain_schema(
+                [
+                    core_schema.datetime_schema(),
+                    core_schema.no_info_plain_validator_function(cls.from_datetime),
+                ]
+            )
+
+            return core_schema.json_or_python_schema(
+                json_schema=from_datetime_schema,
+                python_schema=core_schema.union_schema(
+                    [
+                        core_schema.is_instance_schema(datetime_tz),
+                        from_datetime_schema,
+                    ]
+                ),
```

Comment:
We check whether the value is an instance of `datetime_tz` using [is_instance_schema](https://docs.pydantic.dev/latest/api/pydantic_core_schema/#pydantic_core.core_schema.is_instance_schema), and also apply our `from_datetime_schema` validation.

I based this function a bit on https://python-semver.readthedocs.io/en/latest/advanced/combine-pydantic-and-semver.html

The serialization remains the default `datetime.datetime` serialization